onebusaway-android/src/amazon/java/org/onebusaway/android/map/googlemapsv2/BaseMapFragment.java

```diff
@@ -1030,17 +1030,21 @@ private Dialog createZipCodeDialog() {
                             new DialogInterface.OnClickListener() {
                                 @Override
                                 public void onClick(DialogInterface dialog, int which) {
-                                    String zip = zipInput.getText().toString();
-                                    if (zip != null) {
+                                    String zip = zipInput.getText().toString().trim();
+                                    if (zip != null && LocationUtils.isValidZipCode(zip)) {
                                         String zipCodeKey = getString(R.string.preference_key_zip_code);
                                         PreferenceUtils.saveString(zipCodeKey, zip);
+                                        Location l = LocationUtils.getLocationFromZip(getActivity(), zip);
+                                        if (l != null) {
+                                            Application.setLastKnownLocation(l);
```

Comment:
Hmmm, good question.  Location has always come from an Android `LocationListener` in the past, so we don't have precedence for a location generated by ourselves.

There are a few ways to do this, including injecting the location into `mLocationHelper.onLocationChanged(l)` (which will in turn call `Application.setLastKnownLocation(l)`), or just recentering the map using by setting location `Application.setLastKnownLocation(l)` and then `BaseMapFragment.setMyLocation(true, true`)`.

Let's try injecting the location into the `LocationHelper` in `BaseMapFragment` using `mLocationHelper.onLocationChanged(l);`.  It might just work.